modules/decima-ext-model-exporter/src/main/java/com/shade/decima/ui/data/viewer/model/dmf/DMFExporter.java

```diff
@@ -483,6 +617,59 @@ private DMFNode toModel(
         return res;
     }
 
+    private DMFNode terrainTileDataToModel(
+        @NotNull ProgressMonitor monitor,
+        @NotNull CoreBinary core,
+        @NotNull RTTIObject object,
+        @NotNull String resourceName
+    ) throws IOException {
+        RTTIObject gridCoordinates = object.get("GridCoordinates");
+        Point gridPoint = new Point(gridCoordinates.i32("X"), gridCoordinates.i32("Y"));
+        TileData tileData = tiles.getOrDefault(gridPoint, new TileData(gridPoint));
```

Comment:
Are you sure this won't be called a lot and result in extra redundant allocations?